index.js

```diff
@@ -125,6 +136,459 @@
     return result;
   };
 
+  var strRepeat = function(s, times) {
+    return Array(times + 1).join(s);
+  };
+
+  //  findNodePosition :: Node -> ((Node, [Nodes]) -> Node) -> Position
```

Comment:
If `sig :: a`, we must have `fn :: a -> b` since we have `fn(sig)`. The first `a` is `Node` and the second is `(Node, [Nodes])`, so something is amiss.